packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -82,10 +83,25 @@ class CodeSizeBodyState extends State<CodeSizeBody>
     final newController = Provider.of<CodeSizeController>(context);
     if (newController == controller) return;
     controller = newController;
+
+    snapshotRoot = controller.snapshotRoot.value;
+    addAutoDisposeListener(controller.snapshotRoot, () {
+      setState(() {
+        snapshotRoot = controller.snapshotRoot.value;
+      });
+    });
+
+    diffRoot = controller.diffRoot.value;
+    addAutoDisposeListener(controller.diffRoot, () {
+      setState(() {
+        diffRoot = controller.diffRoot.value;
+      });
+    });
```

Comment:
why did we move this state up the tree? Now if either root changes, both the snapshot and diff view will rebuild (even if it is not the active view)